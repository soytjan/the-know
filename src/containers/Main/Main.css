@import url("https://fonts.googleapis.com/css?family=Open+Sans:300,400|Poiret+One");
.Main {
  background-color: #ECF0F1;
  padding: 0 30px 30px; }
  .Main h3 {
    font-size: 1.5em;
    font-weight: 400;
    padding: 35px;
    text-transform: uppercase; }

.main-footer {
  background-color: #4ECDC4;
  color: #FBFCFC;
  height: 60px;
  display: flex;
  justify-content: center;
  align-items: center; }
  .main-footer p {
    font-size: 1.2em; }

.loading {
  background-image: url("../../assets/loading.gif");
  height: 250px;
  width: 320px;
  margin: auto;
  margin-top: 50px; }

@media screen and (max-width: 588px) {
  .Main h3 {
    padding-bottom: 20px; } }
