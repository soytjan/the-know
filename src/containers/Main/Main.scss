@import "../../variables.scss";

.Main {
  background-color: $main-bg;
  padding: 0 30px 30px;

  h3 {
    font-size: 1.5em;
    font-weight: 400;
    padding: 35px;
    text-transform: uppercase;
  }
}

.main-footer {
  background-color: $turq;
  color: $white;
  height: 60px;
  display: flex;
  justify-content: center;
  align-items: center;

  p {
    font-size: 1.2em;
  }
}

.loading {
  background-image: url('../../assets/loading.gif');
  height: 250px;
  width: 320px;
  margin: auto;
  margin-top: 50px;
}

@media screen and (max-width: 588px) {
  .Main {
    h3 {
      padding-bottom: 20px;
    }
  }
}