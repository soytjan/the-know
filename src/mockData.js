export const mockEventData = {
  "last_item": null,
  "total_items": "9621",
  "first_item": null,
  "page_number": "1",
  "page_size": "10",
  "page_items": null,
  "search_time": "0.049",
  "page_count": "963",
  "events": {
    "event": [
      {
        "watching_count": null,
        "olson_path": "America/Denver",
        "calendar_count": null,
        "comment_count": null,
        "region_abbr": "CO",
        "postal_code": null,
        "going_count": null,
        "all_day": "0",
        "latitude": "39.7392358",
        "groups": null,
        "url": "http://denver.eventful.com/events/thursday-work-day-july-5-multiple-neighborhoods-/E0-001-111209195-8?utm_source=apis&utm_medium=apim&utm_campaign=apic",
        "id": "E0-001-111209195-8",
        "privacy": "1",
        "city_name": "Denver",
        "link_count": null,
        "longitude": "-104.990251",
        "country_name": "United States",
        "country_abbr": "USA",
        "region_name": "Colorado",
        "start_time": "2018-07-05 09:00:00",
        "tz_id": null,
        "description": " <strong>Sponsored by:</strong> Extreme Community Makeover<br> <strong>Minimum age:</strong> 14<br> Extreme Community Makeover (ECM) is designed to invite volunteers into a partnership with residents of specific Denver neighborhoods to complete home and neighborhood improvement projects. These projects include graffiti removal, landscaping, painting, yard work, or other projects identified by the . . .",
        "modified": "2018-02-19 06:12:10",
        "venue_display": "0",
        "tz_country": null,
        "performers": null,
        "title": "Thursday Work Day, July 5 - Multiple Neighborhoods",
        "venue_address": "Denver, Colorado, United States",
        "geocode_type": "EVDB Geocoder",
        "tz_olson_path": null,
        "recur_string": null,
        "calendars": null,
        "owner": "evdb",
        "going": null,
        "country_abbr2": "US",
        "image": null,
        "created": "2018-02-01 04:56:32",
        "venue_id": "V0-001-000179395-7",
        "tz_city": null,
        "stop_time": null,
        "venue_name": "Denver, Colorado, United States",
        "venue_url": "http://denver.eventful.com/venues/denver-colorado-united-states-/V0-001-000179395-7?utm_source=apis&utm_medium=apim&utm_campaign=apic"
      }
    ]
  }
}

export const mockCleanEventData = [
  {
    title: "Thursday Work Day, July 5 - Multiple Neighborhoods",
    description: " <strong>Sponsored by:</strong> Extreme Community Makeover<br> <strong>Minimum age:</strong> 14<br> Extreme Community Makeover (ECM) is designed to invite volunteers into a partnership with residents of specific Denver neighborhoods to complete home and neighborhood improvement projects. These projects include graffiti removal, landscaping, painting, yard work, or other projects identified by the . . .",
    category: "event",
    venueName: "Denver, Colorado, United States", 
    region: "CO",
    postalCode: null,
    city: "Denver",
    startTime: "2018-07-05 09:00:00",
    venueAddress: "Denver, Colorado, United States",
    image: null,
    url: "http://denver.eventful.com/events/thursday-work-day-july-5-multiple-neighborhoods-/E0-001-111209195-8?utm_source=apis&utm_medium=apim&utm_campaign=apic",
    id: "E0-001-111209195-8"
  }
]