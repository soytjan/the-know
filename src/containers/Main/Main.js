import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import {
  updateEvents,
  addFavorite,
  removeFavorite
} from '../../actions/';
import Events from '../../components/Events/Events';
import './Main.css';

export class Main extends Component {
  handleFavorites = (event) => {
    const { favorites, updateEvents } = this.props;
    const favIds = Object.keys(favorites);
    const isDuplicated = favIds.some(id => id === event.id);
    const favEvent = {...event, isFavorited: !event.isFavorited };
    isDuplicated ? this.removeFavEvent(favEvent) : this.addFavEvent(favEvent);

    updateEvents(favEvent);
  }

  // look at this for a refactor 
  addFavEvent = (event) => {
    const { addFavorite } = this.props;

    addFavorite(event); 
  }

  removeFavEvent = (event) => {
    const { removeFavorite } = this.props;

    removeFavorite(event);
  }

  render() {
    const { events, favorites } = this.props;

    if (!events.music) {
      return (
        <div className='loading'></div>
      );
    }

    return (
      <div>
        <section className="Main">
          <h3>{this.props.type}</h3>
          <Route 
            exact path='/home/' 
            render={() => (<Events 
              info={events} 
              type='event' 
              onFavorite={this.handleFavorites} />)}
          />
          <Route 
            exact path='/home/music' 
            render={() => (<Events 
              info={events.music} 
              type='music' 
              onFavorite={this.handleFavorites} />)}
          />
          <Route 
            exact path='/home/food' 
            render={() => (<Events 
              info={events.food} 
              type='food' 
              onFavorite={this.handleFavorites} />)}
          />
          <Route 
            exact path='/home/culture' 
            render={() => (<Events 
              info={events.culture} 
              type='culture' 
              onFavorite={this.handleFavorites} />)}
          />
          <Route 
            exact path='/home/nightlife' 
            render={() => (<Events 
              info={events.nightlife} 
              type='nightlife' 
              onFavorite={this.handleFavorites} />)}
          />
          <Route 
            exact path='/home/favorites' 
            render={() => (<Events 
              info={favorites} 
              type='favorites' 
              onFavorite={this.handleFavorites} />)}
          />
          <Route 
            exact path='/home/search' 
            render={() => (<Events 
              info={events.search} 
              type='search' 
              onFavorite={this.handleFavorites} />)}
          />
        </section>
        <footer className='main-footer'>
          <p>&copy; Amanda Tjan</p>
        </footer>
      </div>
    );
  }
}

Main.propTypes = {
  events: PropTypes.object,
  favorites: PropTypes.object,
  updateEvents: PropTypes.func,
  addFavorite: PropTypes.func,
  removeFavorite: PropTypes.func,
  type: PropTypes.string
};

export const mapStateToProps = (state) => ({
  events: state.events,
  favorites: state.favorites
});

export const mapDispatchToProps = (dispatch) => ({
  updateEvents: event => dispatch(updateEvents(event)),
  addFavorite: event => dispatch(addFavorite(event)),
  removeFavorite: event => dispatch(removeFavorite(event))
});

export default connect(mapStateToProps, mapDispatchToProps)(Main);


        