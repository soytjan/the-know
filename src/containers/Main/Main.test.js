import React from 'react';
import { shallow } from 'enzyme';
import { mockCleanEventData, mockSingleEvent } from '../../mockData';
import { Main, mapStateToProps, mapDispatchToProps } from './Main';

describe('Main', () => {
  let renderedComponent;
  let mockEvents;
  let mockFavorites;
  let mockUpdateEvents;
  let mockAddFavorite;
  let mockRemoveFavorite;

  beforeAll(() => {
    mockEvents = {...mockCleanEventData, '123': {id: '123', category: 'food', title: 'hi'}};
    mockFavorites = {"E0-001-111209195-8": {...mockSingleEvent, isFavorited: true}};
  });

  beforeEach(() => {
    mockUpdateEvents = jest.fn();
    mockAddFavorite = jest.fn();
    mockRemoveFavorite = jest.fn();
    renderedComponent = shallow(
      <Main 
        type='MUSIC'
        events={mockEvents}
        favorites={mockFavorites}
        updateEvents={mockUpdateEvents}
        addFavorite={mockAddFavorite}
        removeFavorite={mockRemoveFavorite}
      />
    );
  });

  it('should match snapshot', () => {
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should map to the store correctly', () => {
    const mockStore = {
      events: mockEvents,
      favorites: mockFavorites
    };
    const mapped = mapStateToProps(mockStore);

    expect(mapped).toEqual(mockStore);
    expect(mapped.events).toEqual(mockEvents);
    expect(mapped.favorites).toEqual(mockFavorites);
  });

  describe('mapDispatchToProps', () => {
    let mockDispatch;
    let mapped;

    beforeEach(() => {
      mockDispatch = jest.fn();
      mapped = mapDispatchToProps(mockDispatch);
    });

    it('should call the dispatch function when updateEvents from MDTP', () => {
      mapped.updateEvents();

      expect(mockDispatch).toHaveBeenCalled();
    });

    it('should call the dispatch function when addFavorite from MDTP', () => {
      mapped.addFavorite();

      expect(mockDispatch).toHaveBeenCalled();
    });

    it('should call the dispatch function when removeFavorite from MDTP', () => {
      mapped.removeFavorite();

      expect(mockDispatch).toHaveBeenCalled();
    });
  });

  describe('handleFavorites', () => {
    it('should check to see if an event in the favorites array, and add it if it is not', () => {
      const mockEvent = {'123': {id: '123', category: 'food', title: 'hi'}};

      renderedComponent.instance().handleFavorites(mockEvent);

      expect(mockAddFavorite).toHaveBeenCalled();
    });

    it('should check to see if an event in the favorites array, and remove it if it is', () => {
      renderedComponent.instance().handleFavorites(mockSingleEvent);

      expect(mockRemoveFavorite).toHaveBeenCalled();
    });

    it('should update events in the master object', () => {
      renderedComponent.instance().handleFavorites(mockSingleEvent);

      expect(mockUpdateEvents).toHaveBeenCalled();
    });
  });

  describe('addFavEvent', () => {
    it('should call addFavorite from store', () => {
      renderedComponent.instance().addFavEvent();

      expect(mockAddFavorite).toHaveBeenCalled();
    });
  });

  describe('removeFavEvent', () => {
    it('should call removeFavorite from store', () => {
      renderedComponent.instance().removeFavEvent();

      expect(mockRemoveFavorite).toHaveBeenCalled();
    });
  });
});