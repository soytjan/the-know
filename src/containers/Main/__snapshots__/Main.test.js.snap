// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    addFavorite={[Function]}
    events={
        Object {
            "123": Object {
              "category": "food",
              "id": "123",
              "title": "hi",
            },
            "E0-001-111209195-8": Object {
              "category": "music",
              "city": "Denver",
              "description": " <strong>Sponsored by:</strong> Extreme Community Makeover<br> <strong>Minimum age:</strong> 14<br> Extreme Community Makeover (ECM) is designed to invite volunteers into a partnership with residents of specific Denver neighborhoods to complete home and neighborhood improvement projects. These projects include graffiti removal, landscaping, painting, yard work, or other projects identified by the . . .",
              "id": "E0-001-111209195-8",
              "image": null,
              "postalCode": null,
              "region": "CO",
              "startTime": "2018-07-05 09:00:00",
              "title": "Thursday Work Day, July 5 - Multiple Neighborhoods",
              "url": "http://denver.eventful.com/events/thursday-work-day-july-5-multiple-neighborhoods-/E0-001-111209195-8?utm_source=apis&utm_medium=apim&utm_campaign=apic",
              "venueAddress": "Denver, Colorado, United States",
              "venueName": "Denver, Colorado, United States",
            },
          }
    }
    favorites={
        Object {
            "E0-001-111209195-8": Object {
              "category": "music",
              "city": "Denver",
              "description": " <strong>Sponsored by:</strong> Extreme Community Makeover<br> <strong>Minimum age:</strong> 14<br> Extreme Community Makeover (ECM) is designed to invite volunteers into a partnership with residents of specific Denver neighborhoods to complete home and neighborhood improvement projects. These projects include graffiti removal, landscaping, painting, yard work, or other projects identified by the . . .",
              "id": "E0-001-111209195-8",
              "image": null,
              "isFavorited": true,
              "postalCode": null,
              "region": "CO",
              "startTime": "2018-07-05 09:00:00",
              "title": "Thursday Work Day, July 5 - Multiple Neighborhoods",
              "url": "http://denver.eventful.com/events/thursday-work-day-july-5-multiple-neighborhoods-/E0-001-111209195-8?utm_source=apis&utm_medium=apim&utm_campaign=apic",
              "venueAddress": "Denver, Colorado, United States",
              "venueName": "Denver, Colorado, United States",
            },
          }
    }
    removeFavorite={[Function]}
    type="MUSIC"
    updateEvents={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "loading",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "loading",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
